### ISSUE TITLE
Server side rendering issue on default route

### ISSUE THREAD

**Issue by scaledrop** [luangjokaj/react-fondue#2]
I would like to highlight one issue which I have faced recently.

Step 1. 
I forked your [repo](https://github.com/luangjokaj/react-ssr-boilerplate/tree/d85f08038e722e6aab8f6c52c21a50514173b1d9)  However, This is little old but I didn't need the locales support in my project. Though I updated all dependencies ;-) 

Step 2.
I replaced the code in App/Routes.js

        <Switch>
            <Redirect exact from="/" to="/home" />
            <Route exact path="/about">
                <UniversalComponent page="About" />
            </Route>
            <Route exact path="/article">
                <UniversalComponent page="Article" />
            </Route>
            <Route exact path="/home">
                <UniversalComponent page="Home" />
            </Route>
            <Route>
                <NotFound />
            </Route>
        </Switch>

with the following one.

    <Switch>
      <Route exact path="/">
        <UniversalComponent page="Home" />
      </Route>
      <Route exact path="/about">
        <UniversalComponent page="About" />
      </Route>
      <Route exact path="/article">
        <UniversalComponent page="Article" />
      </Route>
      <Route>
        <NotFound />
      </Route>
    </Switch>

Step 3.

npm run start

Result:

Basically, I changed the routing for Home page. I want whenever i open http://localhost:8080 in the browser, it should open the Home component. But it is loading NotFound component on the server then it renders the Home component on client side.
I want to load Home when i load the http://localhost:8080 by default, from server and client both.

**Comment by scaledrop** [comment]
@luangjokaj Will you please look into it?

**Comment by luangjokaj** [comment]
@scaledrop can you try to browse it from incognito mode? The code you posted here is working for me there are no issues, just some aggressive caching.

https://i.imgur.com/3BmBEAS.png
https://i.imgur.com/j6OgCxG.png

**Comment by scaledrop** [comment]
@luangjokaj Thank you so much for clarifying that. But the issue is with the **production** build otherwise it is working fine in dev environment. Also, would you please guide me how to remove the locale support without losing anything crucial?

**Comment by scaledrop** [comment]
@luangjokaj Please find the screenshot attached. https://imgur.com/a/Yg8zHwf 
I had no intention to highlight a silly thing here and waste the time of both of us. It is seriously **not rendering from server in production mode**. I hope you understand my concern now.

**Comment by luangjokaj** [comment]
@scaledrop On your **AppRoot.js** remove the following:

```
const isEn = location.pathname.substr(1, 2) == 'en' && 'en';
const isDe = location.pathname.substr(1, 2) == 'de' && 'de';
const currentLang = isEn || isDe || 'en';
```

and also get rid of `lang={navigator && currentLang}`

You can cleanup also `server/render.js` from all the lang references.

**Comment by scaledrop** [comment]
I followed the same steps and got rid of lang from everywhere @luangjokaj But not working. Please see the screenshot once while it was in production env

**Comment by scaledrop** [comment]
@luangjokaj It is strange though. React-universal-component should work properly in case of **"/"** or Home route.

**Comment by luangjokaj** [comment]
@scaledrop remove `exact` from your Home route...

It should look like:
```
<Route
	path="/"
	render={routeProps => <UniversalComponent page="Home" {...routeProps} />}
/>
```

**Comment by scaledrop** [comment]
Oh, it seems to be working without exact, thanks @luangjokaj !

**Comment by scaledrop** [comment]
But now code-splitting stopped working :-/ If i go to another view, its respective CSS and JS don't get loaded. Moreover to this, routing stopped working too :P

**Comment by luangjokaj** [comment]
@scaledrop of course code splitting works only in development, not in production.

**Comment by scaledrop** [comment]
No, then you should go through react-universal-component's [docs](https://github.com/faceyspacey/react-universal-component#intro).  It works on both ends, that's the special ability of it. Did you see, it was working until we removed the exact parameter from Home route!
Now we have to find a way to fix the Home page routing, else is working fine in the boilerplate.

**Comment by scaledrop** [comment]
Any update @luangjokaj ?

**Comment by luangjokaj** [comment]
@scaledrop I just came across this today again. 

I solved it by changing the main home route to: 

```javascript
<Route
	exact
	path="/index.html"
	render={routeProps => <UniversalComponent page="Home" {...routeProps} />}
/>
```

**Comment by scaledrop** [comment]
@luangjokaj Though it's not a proper solution and may vary from host to host, still a nice hack. Thanks man! :+1:
